### DISCUSSION TITLE
LeapFROGS modules

### DISCUSSION THREAD

**Discussion by khider** [LinkedEarth/LeapFROGS#2] [Ideas]
Modules to be considered for LeapFROGS:

- Python
- R
- time series analysis
- github
- docker/Binder/myBinder
- principle of FAIR publishing (geoscience paper of the future)

**Comment by CommonClimate** [comment]
TSA content: 

- harmonic analysis [DAiEES](https://figshare.com/articles/book/Data_Analysis_in_the_Earth_Environmental_Sciences/1014336/11)
- [detrending ](http://linked.earth/PyleoTutorials/notebooks/L1_filtering_and_detrending.html) 
- filtering: [theory](https://figshare.com/articles/book/Data_Analysis_in_the_Earth_Environmental_Sciences/1014336/11) & [practice](http://linked.earth/PyleoTutorials/notebooks/L1_filtering_and_detrending.html) 
- interpolation: [theory](https://figshare.com/articles/book/Data_Analysis_in_the_Earth_Environmental_Sciences/1014336/11) & [practice](http://linked.earth/PyleoTutorials/notebooks/L1_uniform_time_sampling.html)
- spectral analysis [theory](https://figshare.com/articles/book/Data_Analysis_in_the_Earth_Environmental_Sciences/1014336/11) & [practice](http://linked.earth/PyleoTutorials/notebooks/L2_spectral_analysis.html)
- wavelet analysis [theory](https://figshare.com/articles/book/Data_Analysis_in_the_Earth_Environmental_Sciences/1014336/11) & [practice](http://linked.earth/PyleoTutorials/notebooks/L2_wavelet_analysis.html)
- coherency analysis: cross-wavelet [here](http://linked.earth/PyleoTutorials/notebooks/L2_wavelet_analysis.html)
- changepoint detection??

**Reply by DaveEdge1** [reply]
autocorrelation
arima modeling

**Reply by khider** [reply]
- Correlation (which would have autocorrelation)

**Comment by khider** [comment]
Python Modules:

1. Basic Python:
	- https://learnpython.trinket.io/ - Useful for getting started with basic operations. https://docs.trinket.io/getting-started-with-python#/welcome/where-we-ll-go 
		- Exercises (DSCI510):
			- Temperature conversion from Celsius to Fahrenheit
			- Area of a circle given radius
			- Check whether a given year is a leap year
			- Acceptable password
			- Caesar Cypher
	- Need to find information about functions and classes
		- Exercises (DSCI510)
			- Create an Item class
	- Web scrapping? [Low priority]
	- Using Web APIS
		- Exercises (DSCI510)
			- Return weather data
			- Top 3 restaurants from Yelp- Replace with https://waterdata.usgs.gov/blog/api_catalog/
2. Jupyter and Jupyter Notebooks (Already in?)
3. The Scientific Python Stack
	- Project Pythia: https://foundations.projectpythia.org/landing-page.html
		- NumPy
		- Matplotlib
		- Cartopy
		- Datetime -> Will need to be extended
		- Pandas
		- Xarray
	- GeoPandas: https://medium.com/@data95/unlocking-geospatial-data-analysis-with-geopandas-a-comprehensive-python-tutorial-and-analysis-6e131c44dda3
	- Scikit-learn (ML): https://towardsdatascience.com/exploring-use-cases-of-machine-learning-in-the-geosciences-b72ea7aafe2 
		- List of subjects can be found here: https://docs.rcd.clemson.edu/training/workshop_catalog/python/machine_learning_python/

**Comment by khider** [comment]
Publishing resources: 

SPF: https://zenodo.org/records/4744365
FAIR Software: https://fair-software.nl/ / FAIR data: https://www.go-fair.org/fair-principles/ /FAIR workflow: https://workflows.community/groups/fair/ 
EarthCube presentation: [doi.org/10.6084/m9.figshare.25246069](http://doi.org/10.6084/m9.figshare.25246069) 

1. Motivation
2. How to manage data
3. How to manage software
4. How to manage Workflow

Relevant Papers:
1. FAIR data: https://www.nature.com/articles/sdata201618
2. FAIR software: https://www.nature.com/articles/s41597-022-01710-x
3. FAIR workflow: https://direct.mit.edu/dint/article/2/1-2/108/10003/FAIR-Computational-Workflows
